common/src/main/java/dev/latvian/mods/kubejs/recipe/JsonRecipeJS.java

```diff
@@ -35,6 +35,8 @@ public boolean replaceInput(ReplacementMatch match, InputReplacement with) {
 	public boolean hasOutput(ReplacementMatch match) {
 		if (CommonProperties.get().matchJsonRecipes && match instanceof ItemMatch m && getOriginalRecipe() != null) {
 			var r = getOriginalRecipe().getResultItem();
+			//noinspection ConstantValue
+			if (r == null) throw new NullPointerException("ItemStack should never be null, but recipe " + getId() + " returned null as the output!");
```

Comment:
How does this error out during a recipe event? Is it a hard crash or does it continue as usual? Since this is not a user error, we would probably want to make sure it's the latter!

Reply:
It is not a hard crash. From what I can tell it stops the current loop over all recipes (ie that one remove). Any recipes that were iterated before this are still affected.